metricbeat/module/linux/perf/_meta/docs.asciidoc

```diff
@@ -0,0 +1,30 @@
+The `perf` metricsret of the linux module provides insight into running applications via the perf subsystem. This requires elevated permissions.
+Metrics in Perf are divided into two sections: software counters and hardware counters. Hardware counters are usually unavailable on virtualized systems.
+
+
+[float]
+=== Configuration
+
+Perf configuration is based on process globs that are matched to PIDs:
+
+```
+- module: linux
+  period: 10s
+  metricsets:
+     - perf
+  enabled: true
+  #hostfs: /hostfs
+  perf.sample_period: 3s 
+  perf.processes:
+    - process: "systemd"
+      events:
+        hardware: false
+        software: true
+
+```
+
+The `sample_period` metric is universal across all monitored processes. This metric is heavily impacted by the module-wide `period` setting: `perf.sample_period` is a fraction of `linux.period`. 
+In the above, the module `period` setting is 10 seconds, and the `sample_period` is 3 seconds. This means that 3 seconds of performace counters will be gathered every 7 seconds. if `period` is equal to `sample_period`, the processes will be monitored continiously, leading to higher overhead.
```

Comment:
Just want to double check this statement:

> In the above, the module `period` setting is 10 seconds, and the `sample_period` is 3 seconds. This means that 3 seconds of performace counters will be gathered every 7 seconds.

Given that `Fetch()` is called every 10 seconds, and in non-continuous mode, sampling will take 3 seconds (due to the 3-second sleep), isn't it true that "3 seconds of performance counters will be gathered every 10 seconds"?

Reply:
> and in non-continuous mode

Not sure what you're referring to here? Is there some metricset mode switch I don't know about?

Reply:
Nah, I just meant when continuous sampling isn't going on.

Reply:
Oh, I see. My understanding was that libbeat calls `Fetch` on an asynchronous timer, so we're already counting down from the time `Fetch` is called, _not_ once fetch is returned?

Reply:
That's correct. The timer loop for `Fetch` is over here: https://github.com/elastic/beats/blob/95626b8f1690344312c0831ab2bdcbccffe4d089/metricbeat/mb/module/wrapper.go#L226-L236.

A timer based on the module's period is created, and then on every tick of the timer, a new `Fetch` call is made (indirectly, via `msw.fetch`).

Say the `period` is set to 10 seconds and `sample_period` is set to 3 seconds. That means `Fetch` is called every 10 seconds. The first (roughly) 3 seconds of that period is spent gathering samples, but we it's still 10 seconds from one start-of-sampling to the next start-of-sampling. That's why IMO it should say that performance counters will be gathered every 10 seconds.